homeassistant/components/plugwise/climate.py

```diff
@@ -66,20 +66,18 @@ def __init__(
             self._attr_supported_features |= ClimateEntityFeature.PRESET_MODE
             self._attr_preset_modes = presets
 
-        # Determine hvac modes and current hvac mode
-        self._attr_hvac_modes = [HVACMode.HEAT]
-        if self.coordinator.data.gateway["cooling_present"]:
-            self._attr_hvac_modes = [HVACMode.HEAT_COOL]
-        if self.device["available_schedules"] != ["None"]:
-            self._attr_hvac_modes.append(HVACMode.AUTO)
-
         self._attr_min_temp = self.device["thermostat"]["lower_bound"]
         self._attr_max_temp = self.device["thermostat"]["upper_bound"]
         # Ensure we don't drop below 0.1
         self._attr_target_temperature_step = max(
             self.device["thermostat"]["resolution"], 0.1
         )
 
+        # Determine stable hvac_modes
+        self._hvac_modes: list[HVACMode] = [HVACMode.HEAT]
+        if self.coordinator.data.gateway["cooling_present"]:
+            self._hvac_modes = [HVACMode.HEAT_COOL]
```

Comment:
Let's move this part into the property method as well at this point.

Reply:
Tnx for reviewing, I wasn't sure what would be the preferred way to do this, will change this later today.

Reply:
Implemented.